File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        k = 0 # array position marker (slow runner - where next unique elem placed)
        for x in nums:
            # Check if k is 0 --> we're at the start of the array
            # or if the current number is not equal to the last unique number we've seen.
            # The condition k == 0 is true for the first element, so we store it as the first unique element.
            if k == 0 or x != nums[k - 1]:
                nums[k] = x # current unique elem at the k-th position in array
                k += 1 # increment to indicate next unique elem placed in next position
        # Return the length of the array containing all unique elements (new length of the array without duplicates)
        return k

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let k = 0 // tracks the position where the next unique elem should be placed in array

    // Iterate Through the array of nums - k(slow runner), x(fast runner)
    for(const x of nums){
        // Identify Unique Elements
        // Check if at the start of the array (k === 0) or the current elem is not the same as the last unique elem(then unique)
        if (k === 0 || x !== nums[k - 1]){
            // If found, store the current elem(x) at the next unique elem position
            nums[k] = x
            k++; // increment the index for the next unique element
        }
        // If the current element is the same as the last unique element,
        // the loop continues without doing anything, effectively skipping duplicates.
    }
    // Return the new length of the array after all duplicates have been removed.
    // This length is equivalent to the next unique element's index.
    return k;
};